pkg/backends/oauth2/groups.go

```diff
@@ -70,5 +83,22 @@ func (b *Backend) fetchUserGroups(tokenData, userData map[string]interface{}) er
 		zap.String("url", userURL),
 	)
 
+	switch b.Config.Provider {
+	case "google":
+		var respParsed googleResponse
+		err = json.Unmarshal(respBody, &respParsed)
+		if err != nil {
+			return err
+		}
+		userGroups := []string{}
+		for _, group := range respParsed.Response.Groups {
+			userGroups = append(userGroups, group.DisplayName)
+		}
+		userData["groups"] = userGroups
```

Comment:
@silvajp2 , the `userData` map making it all the way to the below:

https://github.com/greenpau/caddy-auth-portal/blob/f56ad05f79e9cb4a35495bf730444afa8e72b45b/pkg/authn/handle_external_login.go#L60-L69

Subsequently, it (`rr.Response.Payload`) is being used in here:

https://github.com/greenpau/caddy-auth-portal/blob/f56ad05f79e9cb4a35495bf730444afa8e72b45b/pkg/authn/handle_http_login.go#L118-L151

the `userData["groups"] = userGroups` should really be `userData["roles"] = userGroups`

https://github.com/greenpau/caddy-auth-portal/blob/f56ad05f79e9cb4a35495bf730444afa8e72b45b/pkg/authn/handle_http_login.go#L312

Futher, perhaps, `userData["roles"]` already exists for some reason. Thus, check for it and do not overwrite it. i.e. if exists, append; otherwise, overwrite.